### ISSUE TITLE
Allow mpbuild to run when pwd is not the root of a MicroPython clone

### ISSUE THREAD

**Issue by hmaerki** [mattytrentini/mpbuild#26]
## Bug

```python
mpbuild/src/mpbuild/__init__.py

try:
    os.stat("./ports")
except OSError:
    raise SystemExit("Please run from root of micropython source tree")
```

When I `import pyboard` in octoprobe, I get above exception.

## Proposed fix

Do not throw the exception while importing the module. Throw the exception in the command line module.

Pass the *root of micropython source tree* to `build.build_board()` and other some functions.

This will allow to use mpbuild without the need to `cd`.

**Comment by Josverl** [comment]
I think an exception is fine, but perhaps not a SystemExit.

**Comment by hmaerki** [comment]
> I think an exception is fine, but perhaps not a SystemExit.
This implies, that mpbuild may not be used as a python package (just as external program)...

Again: I can not see any reason why to do do an assertion on the current directory while module loading.

Do you want me to create a PR which fixes this?

**Comment by mattytrentini** [comment]
This has been partially resolved with #28 but I want to take it a little further before closing this issue.

**Comment by hmaerki** [comment]
IMHO, this issue is completely resoved by https://github.com/mattytrentini/mpbuild/pull/54
